frontend/src/metabase/components/Badge/Badge.styled.tsx

```diff
@@ -41,9 +40,7 @@ export const MaybeLink = styled(RawMaybeLink)`
   }
 `;
 
-export const BadgeIcon = styled(Icon, {
-  shouldForwardProp: shouldForwardNonTransientProp,
-})<{ $hasMargin: boolean }>`
+export const BadgeIcon = styled(Icon)<{ $hasMargin: boolean }>`
```

Comment:
nit: you can rename prop from `$hasMargin` to `hasMargin` as everywhere

Reply:
Fixed